rust/cubesqlplanner/cubesqlplanner/src/planner/sql_evaluator/sql_call.rs

```diff
@@ -217,16 +220,72 @@ impl SqlCall {
                     query_tools.base_tools().security_context_for_rust()?,
                 ))
             }
-            ContextSymbolDep::FilterParams => MemberSqlArg::ContextSymbol(
-                ContextSymbolArg::FilterParams(query_tools.base_tools().filters_proxy()?),
-            ),
-            ContextSymbolDep::FilterGroup => MemberSqlArg::ContextSymbol(
-                ContextSymbolArg::FilterGroup(query_tools.base_tools().filter_group_function()?),
-            ),
+            ContextSymbolDep::FilterParams => {
+                let filters = visitor.all_filters();
+                let native_filters = self.filters_to_native_filter_item(filters);
+                let r = query_tools
+                    .base_tools()
+                    .filters_proxy_for_rust(native_filters)?;
+                MemberSqlArg::ContextSymbol(ContextSymbolArg::FilterParams(r))
+            }
+            ContextSymbolDep::FilterGroup => {
+                let filters = visitor.all_filters();
+                let native_filters = self.filters_to_native_filter_item(filters);
+                let r = query_tools
+                    .base_tools()
+                    .filter_group_function_for_rust(native_filters)?;
+                MemberSqlArg::ContextSymbol(ContextSymbolArg::FilterGroup(r))
+            }
             ContextSymbolDep::SqlUtils => MemberSqlArg::ContextSymbol(ContextSymbolArg::SqlUtils(
                 query_tools.base_tools().sql_utils_for_rust()?,
             )),
         };
         Ok(res)
     }
+
+    fn filters_to_native_filter_item(
+        &self,
+        filter: Option<Filter>,
+    ) -> Option<Vec<NativeFilterItem>> {
+        if let Some(filter) = filter {
+            let mut res = Vec::new();
+            for item in filter.items.iter() {
+                res.push(self.filters_to_native_filter_item_impl(item));
+            }
+            Some(res)
+        } else {
+            None
+        }
+    }
+
+    fn filters_to_native_filter_item_impl(&self, filter_item: &FilterItem) -> NativeFilterItem {
+        match filter_item {
+            FilterItem::Group(group) => {
+                let mut native_items = Vec::new();
+                for itm in group.items.iter() {
+                    native_items.push(self.filters_to_native_filter_item_impl(itm));
+                }
```

Comment:
Same here, more idiomatic style:
```suggestion
let native_items: Vec<_> = group.items
    .iter()
    .map(|itm| self.filters_to_native_filter_item_impl(itm))
    .collect();
```

Reply:
Some as above